core/kernel/persistence/smoothsql/class.Resource.php

```diff
@@ -838,22 +843,41 @@ private function buildTrippleArray(
     private function normalizePropertyValues($value)
     {
         $normalizedValues = [];
-        if ($value instanceof core_kernel_classes_Resource) {
-            $normalizedValues[] = $value->getUri();
-        } elseif (is_array($value)) {
+
+        if (is_array($value)) {
             foreach ($value as $val) {
-                if ($val !== null) {
-                    $normalizedValues[] = $val instanceof core_kernel_classes_Resource
-                        ? $val->getUri()
-                        : $val;
+                if ($val === null) {
+                    continue;
                 }
+                $normalizedValues[] = $this->normalizeSinglePropertyValue($val);
             }
         } else {
-            $normalizedValues[] = ($value == null) ? '' : $value;
+            $normalizedValues[] = $this->normalizeSinglePropertyValue($value);
         }
+
         return $normalizedValues;
     }
 
+    private function normalizeSinglePropertyValue($value)
+    {
+        if ($value === null) {
+            return '';
+        }
+
+        if ($value instanceof core_kernel_classes_Resource) {
+            return $value->getUri();
+        }
+
+        if (is_string($value)) {
+            if (common_Utils::isUri($value)) {
+                return $value;
+            }
+            return helpers_ContentSanitizer::sanitize($value);
+        }
+
+        return $value;
```

Comment:
what with other values like object of different class than core_kernel_classes_Resource ? You will just return them without checking its value.

Reply:
core_kernel_classes_Resource its parent class for all resources in TAO. It should already all of them

Reply:
keep in mind that its how it was before. In last commit we just will check and sanitize string for labels